sourcegraph/sourcegraph
# CodeMirror: make Cmd-f trigger search globally

## Description
As we're approaching shipping CodeMirror-based blob view for everyone, we need to take care of Cmd/Ctrl-F behaviour.
This diff adds it:
- as a global keyboard shortcut to `Layout` - theoretically, it should trigger search everywhere in the app, when the focus isn't hijacked and code mirror blob view is visible
- to the SearchInput, as it has it's own hotkey handling - and we're triggering the CodeMirror search instead

Whenever search triggers, I am actually pressing `Cmd-F`:

https://user-images.githubusercontent.com/2196347/186660156-43a7e00a-5c2b-4ec8-8d9c-f35e61498abf.mov



## Test plan

0. Run the app locally 
1. Set `enableCodeMirrorFileView` to `true` in settings
2. Check that with the focus inside the codemirror view, inside the search bar, or anywhere on the page Cmd-F triggers blob view search
3. Check that outside the page with the blob view, it triggers browser's native search (unless hijacked by some other component)
4. Toggle the setting off and check that the search isn't broken

## App preview:

- [Web](https://sg-web-og-cm-trigger-search.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-fkcmdfctag.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
